src/compo/pandora_2ioda.py

```diff
@@ -0,0 +1,299 @@
+        self.AttrData['Vertice'] = np.int32(self.DimDict['Vertice'])
+
+        varname = 'pressureVertice'
+        vkey = (varname, 'RetrievalAncillaryData')
+        self.varAttrs[vkey]['coordinates'] = 'longitude latitude'
+        self.varAttrs[vkey]['units'] = 'Pa'
+
+    def make_dictionaries(self):
+        """
+        Make all the necessary dictionaries for this class object.
+        """
+
+        self.outData = defaultdict(lambda: DefaultOrderedDict(OrderedDict))
+
+        self.make_obsVars()
+        self.make_AttrData()
+        self.make_varDict()
+        self.make_varAttrs()
+
+    def make_obsVars(self):
+        """
+        Make a dictionary of obsvars
+        """
+        obsvars = {"nitrogendioxideTotal"}
+        self.obsvars = obsvars
+
+    def make_AttrData(self):
+        """
+        Make a dictionary of AttrData based on obsvars
+        """
+        AttrData = {
+            'converter': os.path.basename(__file__),
+            'nvars': np.int32(len(self.obsvars))
+        }
+        self.AttrData = AttrData
+
+    def make_varDict(self):
+        """
+        """
+        self.varDict = defaultdict(lambda: defaultdict(dict))
+        for item in self.obsvars:
+            self.varDict[item]['valKey'] = item, iconv.OvalName()
+            self.varDict[item]['errKey'] = item, iconv.OerrName()
+            self.varDict[item]['qcKey'] = item, iconv.OqcName()
+
+    def make_varAttrs(self):
+        """
+        Assign attribute to each self.obsvar
+        """
+        self.varAttrs = DefaultOrderedDict(lambda: DefaultOrderedDict(dict))
+        for item in self.obsvars:
+            self.varAttrs[item, iconv.OvalName()]['units'] = 'mol m-2'
+            self.varAttrs[item, iconv.OerrName()]['units'] = 'mol m-2'
+            self.varAttrs[item, iconv.OqcName()]['units'] = 'unitless'
+
+
+def get_parser():
+    """
+    Get the parser object for this script.
+    Returns:
+        parser (ArgumentParser): ArgumentParser which includes all the parser information.
+    """
+
+    # get command line arguments
+    parser = argparse.ArgumentParser(
+        description=(
+            'Reads Pandora L2 total columns of NO2 from txt files (L2_rnvs3)'
+            'and converts into IODA formatted output files. Multiple'
+            'files are able to be concatenated.'),
+        formatter_class=argparse.RawDescriptionHelpFormatter
+    )
+    parser.print_usage = parser.print_help
+
+    required = parser.add_argument_group(title='required arguments')
+    required.add_argument(
+        '-i', '--input',
+        help="path of Pandora measurement txt file",
+        type=str,
+        nargs='+',
+        required=True)
+
+    required.add_argument(
+        '-o', '--output',
+        help="path of IODA output file",
+        type=str,
+        required=True)
+
+    optional = parser.add_argument_group(title='optional arguments')
+    optional.add_argument(
+        '-r', '--time_range',
```

Comment:
```suggestion
    required.add_argument(
        '--time_range',
```

this actually looks like a required argument